lib/Extractor/Candidates.cpp

```diff
@@ -492,6 +497,11 @@ Inst *ExprBuilder::get(Value *V) {
   if (!E) {
     E = build(V);
   }
+  E->DemandedBitsVal = APInt(E->Width, 0, false);
+  if (HarvestDemandedBits)
+    if (Instruction *I = dyn_cast<Instruction>(V))
+      if (DB && (DB->getDemandedBits(I)).getBoolValue())
```

Comment:
I did a null pointer checking because last year when I wrote this patch and tested it to build llvm-3.8 (if i remember correctly), it crashed giving me some errors related to this. So, I added this check. I should test it without null pointer checking against llvm-5.0.0. 
The other check on demanded bits value is added to make sure we only store the live demanded bits only, not the dead demanded bits value. I can see this piece of code looks too redundant.